src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -354,9 +355,16 @@ static int add_swap(const char *path) {
                 return 0;
         }
 
-        log_debug("Adding swap: %s", path);
+        if (streq_ptr(p->fstype, "crypto_LUKS")) {
+                r = add_cryptsetup("swap", p->node, true, true, &what);
+                if (r < 0)
+                        return r;
+        } else
+                what = p->node;
```

Comment:
This will get freed when exiting the block by _cleanup_free_. Have an auxiliary `const char *path` that gets assigned the value of `what` in the above block instead, to avoid that, and use that below instead of `what`

Reply:
Wouldn't that mean it's freed after the function returns? I don't see any lingering references to it.

Reply:
Yeah, we don't want to free `p->node` here, this function doesn't own it

Reply:
Aah, now I get it!

Reply:
Does 37df1cc94e look okay?

Reply:
yeah, looks good - can you please squash all commits and force push?